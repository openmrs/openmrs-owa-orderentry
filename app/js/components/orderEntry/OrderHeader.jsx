import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const OrderHeader = ({ status, orderable }) => (
  <Fragment>
    <th className="th-invisible">{orderable}</th>
    <th className="th-invisible">{status}</th>
    <th className="th-invisible">
      <i className="icon-edit" />
      <i className="icon-delete" />
    </th>
  </Fragment>
);

export default OrderHeader;

OrderHeader.defaultProps = {
  status: '',
  orderable: '',
};

OrderHeader.propTypes = {
  status: PropTypes.string,
  orderable: PropTypes.string,
};
