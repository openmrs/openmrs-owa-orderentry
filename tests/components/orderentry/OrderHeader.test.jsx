import React from 'react';
import OrderHeader from '../../../app/js/components/orderEntry/OrderHeader';

let props;
let mountedComponent;

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<OrderHeader {...props} />);
  }
  return mountedComponent;
};

describe('Component: OrderHeader', () => {
  beforeEach(() => {
    props = {
      status: 'iactive',
      orderable: 'some oderable',
    };
    mountedComponent = undefined;
  });

  it('renders properly', () => {
    const component = getComponent();
    expect(component).toMatchSnapshot();
  });
});
