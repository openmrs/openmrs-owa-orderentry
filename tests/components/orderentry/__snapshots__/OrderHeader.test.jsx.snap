// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: OrderHeader renders properly 1`] = `
<React.Fragment>
  <th>
    some oderable
  </th>
  <th>
    iactive
  </th>
  <th
    className="order-actions-btn"
  >
    <IconButton
      dataContext={Object {}}
      iconClass="icon-pencil"
      iconTitle="EDIT"
      onClick={[Function]}
    />
    <IconButton
      dataContext={Object {}}
      iconClass="icon-remove"
      iconTitle="DISCONTINUE"
      onClick={[Function]}
    />
  </th>
</React.Fragment>
`;
